merlin_models/tf/block/retrieval.py

```diff
@@ -100,14 +100,12 @@ def TwoTowerBlock(
 def MatrixFactorizationBlock(
     schema: Schema,
     dim: int,
-    query_id_tag=Tag.USER_ID,
-    item_id_tag=Tag.ITEM_ID,
+    query_id_tag=Tags.USER_ID,
+    item_id_tag=Tags.ITEM_ID,
     embeddings_initializers: Optional[Dict[str, Callable[[Any], None]]] = None,
     **kwargs,
 ):
-    query_item_schema = schema.select_by_tag(
-        lambda tags: query_id_tag in tags or item_id_tag in tags
-    )
+    query_item_schema = schema.select_by_tag(query_id_tag) + schema.select_by_tag(item_id_tag)
```

Comment:
This seems like a reasonable way to implement the same behavior on the other version of the `Schema` API. I do wonder if we should think about integrating the `lambda` functionality into core though.